### ISSUE TITLE
Rendering issues with animated BlockEntityGenericTypedContainer

### ISSUE THREAD

**Issue by DArkHekRoMaNT** [anegostudios/VintageStory-Issues#1882]
**Game Version:** 1.16.5

### Description
Rendering issues in BlockEntityGenericTypedContainer when creating custom containers with the same shape and different types (in attributes). When caching, the type is ignored, shapename is used instead.

```
if (animUtil != null)
{
    string skeydict = "typedContainerShapes";
    Dictionary<string, Shape> shapes = ObjectCacheUtil.GetOrCreate(Api, skeydict, () =>
    {
        return new Dictionary<string, Shape>();
    });
    string skey = Block.FirstCodePart() + block.Subtype + "-" + "-" + shapename + "-" + rndTexNum;
    if (!shapes.TryGetValue(skey, out shape))
    {
        shapes[skey] = shape = block.GetShape(Api as ICoreClientAPI, type, shapename, tesselator, rndTexNum);
    }
}
```
skey example here: "wchest-north--game:block/wood/chest-0"

### Expected behavior
Correct rendering depending on chest type

### Possible solution
Replace shapename with type (or add) in skey ([here](https://github.com/anegostudios/vssurvivalmod/blob/c1961caf8fecb4a463f34a345206058decde7205/BlockEntity/BEGenericTypedContainer.cs#L331))

### Screenshots
![](https://media.discordapp.net/attachments/923237597831122996/970246884306337822/unknown.png)

**Comment by DArkHekRoMaNT** [comment]
Related to #1550

**Comment by Craluminum2413** [comment]
Fixed in [1.18.0-pre.4](https://github.com/anegostudios/vssurvivalmod/blob/fecc4d6c50c74e08a5dc7869eff7f967bc2577ac/BlockEntity/BEGenericTypedContainer.cs#L331)
